code/modules/projectiles/guns/chaos_bolt.dm

```diff
@@ -0,0 +1,377 @@
+	var/obj/item/item_to_summon
+	/// If left at 0, item goes in backpack or floor, if set, throw that many items around the target.
+	var/explosion_amount = 0
+	/// Name of random effect to be applied on target mob.
+	var/chaos_effect
+
+/obj/item/projectile/magic/chaos/on_hit(atom/target, blocked = 0)
+	. = ..()
+
+	if(iswallturf(target) || isobj(target))
+		target.color = pick(GLOB.random_color_list)
+		return
+
+	if(target && isliving(target))
+		var/mob/living/L = target
+		if(L.stat == DEAD)
+			L.visible_message("<span class='warning'>[target] glows faintly, but nothing else happens.</span>")
+			return
+		chaos_chaos(L)
+
+/obj/item/projectile/magic/chaos/Initialize(mapload)
+	. = ..()
+	icon_state = pick("bluespace", "pulse1", "magicm", "declone", "fireball", "blood_bolt", "arcane_barrage", "laser", "u_laser")
+
+
+ /**
+  * Picks and call a subproc to apply a random effect on mob/living/target.
+  *
+  * First pick a category of random effect,
+  * then calls a sub-proc to pick and apply an effect in that category,
+  * then summons any item_to_summon set by effects.
+  * Arguments:
+  * * target - mob/living that will have effect applied on them
+  */
+/obj/item/projectile/magic/chaos/proc/chaos_chaos(mob/living/target)
+	var/category = pick(prob(CHAOS_STAFF_LETHAL_CHANCE);"lethal", prob(CHAOS_STAFF_NEGATIVE_CHANCE);"negative", prob(CHAOS_STAFF_MISC_CHANCE);"misc",\
+		prob(CHAOS_STAFF_GIFT_CHANCE);"gift", prob(CHAOS_STAFF_GREAT_GIFT_CHANCE);"great gift")
+	switch(category)
+		if("lethal") //Target is either dead on the spot or might as well be
+			apply_lethal_effect(target)
+		if("negative") //Target is damaged, crippled or otherwise negatively affected. Effect can be lethal in some circumstances.
+			apply_negative_effect(target)
+		if("misc") //Miscellaneous effect, can be positive, negative, or just humorous
+			apply_misc_effect(target)
+		if("gift") //Grants a gift or positive effect to the target. Usually a gag or mildly useful item... if you weren't being killed by a wizard
+			apply_gift_effect(target)
+		if("great gift") //Grants a gift or positive effect to the target. Usually a weapon or useful item.
+			apply_great_gift_effect(target)
+	if(!item_to_summon)
+		return
+	if(!target.mind) //no abusing mindless mobs for free stuff
+		target.visible_message("<span class='warning'>[target] glows faintly, but nothing else happens.</span>")
+		return
+	if(explosion_amount)
+		target.visible_message("<span class='chaosneutral'>A bunch of [item_to_summon.name] scatter around [target]!</span>", \
+			"<span class='chaosneutral'>A bunch of [item_to_summon.name] scatter around you!</span>")
+		for(var/i in 1 to explosion_amount)
+			var/obj/item/I = new item_to_summon(get_turf(target))
+			throwforce = 0
+			INVOKE_ASYNC(I, TYPE_PROC_REF(/atom/movable, throw_at), pick(oview(7, get_turf(src))), 10, 1)
+			throwforce = initial(throwforce)
+	else
```

Comment:
No need for things below this to be in an else block, just return after the for loop (one level up from it) and un-indent the else block